src/NLU.DevOps.ModelPerformance/TestCaseSource.cs

```diff
@@ -71,12 +71,15 @@ string getUtteranceId(LabeledUtterance utterance, int index)
                 return utterance.GetUtteranceId() ?? index.ToString(CultureInfo.InvariantCulture);
             }
 
+            var intents = actualUtterances.Select(utterance => utterance.Intent).Distinct().ToList();
```

Comment:
This is not an accurate representation of all possible intents. The only source of truth for all possible intents is the model itself. I think we may want to take this library out of the business of determining true negatives.

Reply:
yeah it is not always going to be accurate. I was assuming test cases would cover all intents but that is not a guarantuee

Reply:
good point. I was assuming the test set would include all intents but this is not guarantueed and the predictioons are not guarantuees to contain all intents.